### PR TITLE
Added method to log long messages to AI

### PR DESCRIPTION
**Pull request by dorfire**

### CODE REVIEW

File: src/Common/Extensions/LoggerAiExtensions.cs [hunk 1]
```diff
@@ -43,6 +48,24 @@ public static void LogTrace(this ILogger logger, string message, IDictionary<str
...
+        /// <param name="header">chunk header. will be followed by chunk index, a colon, and a line break</param>
+        /// <param name="message">long message to be chunkified and logged</param>
+        public static void LogLongMessage(this ILogger logger, LogLevel level, [NotNull] string header, string message)
+        {
+            int maxChunkLen = AiMessageLengthLimit - header.Length - 16; // 16 should cover for the header formatting
+            int i = 0;
+            var chunks = Chunkify(message, maxChunkLen);
+            foreach (string chunk in chunks)
+            {
+                logger.Log(level, $"{header} ({++i}/{chunks.Count}):\n{chunk}");
```
Comment (kichalla) [inline src/Common/Extensions/LoggerAiExtensions.cs]: 👍 for `({++i}/{chunks.Count})`

File: tests/Oryx.Common.Test/LoggerAiExtensionsTest.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// --------------------------------------------------------------------------------------------
+
+using Xunit;
+
+namespace Microsoft.Extensions.Logging
+{
+    public class LoggerAiExtensionsTest
+    {
+        [Fact]
+        public void SplitByLength_Sanity()
```
Comment (kichalla) [inline tests/Oryx.Common.Test/LoggerAiExtensionsTest.cs]: 👍

File: src/Common/Extensions/LoggerAiExtensions.cs [hunk 3]
```diff
@@ -43,6 +48,24 @@ public static void LogTrace(this ILogger logger, string message, IDictionary<str
...
+        /// Logs a long message in chunks, with each chunk limited in length to 2^15.
+        /// </summary>
+        /// <param name="logger">logger instance</param>
+        /// <param name="level">log level. will apply to all chunks</param>
+        /// <param name="header">chunk header. will be followed by chunk index, a colon, and a line break</param>
+        /// <param name="message">long message to be chunkified and logged</param>
+        public static void LogLongMessage(this ILogger logger, LogLevel level, [NotNull] string header, string message)
+        {
+            int maxChunkLen = AiMessageLengthLimit - header.Length - 16; // 16 should cover for the header formatting
+            int i = 0;
```
Comment (kichalla) [inline src/Common/Extensions/LoggerAiExtensions.cs]: nit: since `chunks` is a `IList`, you can use indexes and avoid this

Comment (dorfire) [reply src/Common/Extensions/LoggerAiExtensions.cs]: true, but I think it's cleaner with `foreach` rather than `for`, WDYT?

**Pull request merged**